core/lib/zksync_core/src/api_server/execution_sandbox/validate.rs

```diff
@@ -72,36 +73,39 @@ impl TxSharedArgs {
                 tx,
                 block_args,
                 |vm, tx| {
-                    let stage_started_at = Instant::now();
+                    let stage_latency = SANDBOX_METRICS.sandbox[&SandboxStage::Validation].start();
                     let span = tracing::debug_span!("validation").entered();
                     vm.push_transaction(&tx);
 
-                    let (tracer, validation_result) = ValidationTracer::<HistoryDisabled>::new(
-                        validation_params,
-                    );
+                    let (tracer, validation_result) =
+                        ValidationTracer::<HistoryDisabled>::new(validation_params);
 
                     let result = vm.inspect_next_transaction(vec![
                         tracer.into_boxed(),
-                        StorageInvocations::new(execution_args.missed_storage_invocation_limit).into_boxed()
+                        StorageInvocations::new(execution_args.missed_storage_invocation_limit)
+                            .into_boxed(),
                     ]);
 
                     let result = match (result.result, validation_result.get()) {
-                        (ExecutionResult::Halt { reason }, _) => Err(ValidationError::FailedTx(reason)),
+                        (ExecutionResult::Halt { reason }, _) => {
+                            Err(ValidationError::FailedTx(reason))
+                        }
                         (_, Some(err)) => Err(ValidationError::ViolatedRule(err.clone())),
                         (_, None) => Ok(()),
                     };
 
-
-                    metrics::histogram!("api.web3.sandbox", stage_started_at.elapsed(), "stage" => "validation");
+                    stage_latency.observe();
                     span.exit();
                     result
                 },
             );
             span.exit();
             result
-        }).await.unwrap();
+        })
+        .await
+        .unwrap();
 
-        metrics::histogram!("server.api.validation_sandbox", stage_started_at.elapsed(), "stage" => "validate_in_sandbox");
+        stage_latency.observe(); // FIXME: renamed from `server.api.validation_sandbox`
```

Comment:
So, instead of a separate metric, this is now a part of the `api_web3_sandbox` metric with the `stage="validate_in_sandbox"` label.